### PR TITLE
Add test case for AndroidView clipBehavior

### LINKED ISSUE #67455: Create a test for RenderAndroidView's clip logic
To avoid future bug like https://github.com/flutter/flutter/pull/67446

### PR DESCRIPTION
**Pull request by xu-baolin**
## Description

@liyuqian 

Add a test case for AndroidView clipBehavior

## Related Issues

Fixes #67455 
#67343 

## Tests

See files.

### CODE REVIEW

File: packages/flutter/test/widgets/platform_view_test.dart [hunk 1]
```diff
@@ -1149,6 +1149,44 @@ void main() {
...
+            child: AndroidView(
+              viewType: 'webview',
+              layoutDirection: TextDirection.ltr,
+              clipBehavior: clipBehavior,
+            ),
+          ),
+        );
+      }
+
+      await tester.pumpWidget(buildView(200.0, 200.0, Clip.none));
```
Comment (liyuqian) [inline packages/flutter/test/widgets/platform_view_test.dart:1169]: This test seems to rely on `RenderAndroidView` to be retained in its last size while the parent widget resizes. I can't find any description about this behavior in https://master-api.flutter.dev/flutter/widgets/AndroidView-class.html. CC @cyanglaz @amirh to double check if this is expected and whether we can rely on this.

File: packages/flutter/test/widgets/platform_view_test.dart [hunk 2]
```diff
@@ -1149,6 +1149,44 @@ void main() {
...
+      // Resize the view.
+      await tester.pumpWidget(buildView(100.0, 100.0, Clip.none));
+      // No clip happen when the clip behavior is `Clip.none` .
+      expect(tester.layers.whereType<ClipRectLayer>(), hasLength(0));
+
+      // No clip happen when the clip behavior only change.
```
Comment (liyuqian) [inline packages/flutter/test/widgets/platform_view_test.dart:1175]: `// No clip happen when the clip behavior only change.` -> `// No clip when only the clip behavior changes while the size remains the same.`

**Review by liyuqian — COMMENTED** [review]
Thanks for writing this test! Here are a few minor comments.

File: packages/flutter/test/widgets/platform_view_test.dart (hunk 1 above)
Comment (xu-baolin) [reply packages/flutter/test/widgets/platform_view_test.dart:1169]: There is an existing test case to test this behavior:
https://github.com/flutter/flutter/blob/4cfd7d3e61022e9b01a9034fc01d8d7aa8a2a967/packages/flutter/test/widgets/platform_view_test.dart#L106-L154

**Review by liyuqian — APPROVED** [review]
LGTM

**Pull request merged**